proxy/src/main/java/com/wavefront/agent/Utils.java

```diff
@@ -31,4 +37,34 @@ public T get() {
       }
     };
   }
+
+  /**
+   * Requires an input uuid string Encoded as 32 hex characters.
+   * For example {@code cced093a76eea418ffdc9bb9a6453df3}
+   *
+   * @param uuid string encoded as 32 hex characters.
+   * @return uuid string encoded in 8-4-4-4-12 (rfc4122) format.
+   */
+  public static String addHyphensToUuid(String uuid) {
+    Matcher matcherUuid = patternUuid.matcher(uuid);
+    return matcherUuid.replaceAll("$1-$2-$3-$4-$5");
+  }
+
+  /**
+   * Method converts a string Id to {@code UUID}.
+   * This Method specifically converts id's with less than 32 digit hex characters into UUID
+   * format (See <a href="http://www.ietf.org/rfc/rfc4122.txt"> <i>RFC&nbsp;4122: A
+   * Universally Unique IDentifier (UUID) URN Namespace</i></a>) by left padding id with Zeroes
+   * and adding hyphens. It assumes that if the input id contains hyphens it is already an UUID.
+   * Please don't use this method to validate/guarantee your id as an UUID.
+   *
+   * @param id a string encoded in hex characters.
+   * @return a UUID string.
+   */
+  public static String convertToUuidString(String id) {
```

Comment:
if input id is Nullable (i.e. line 65), then mark it as @Nullable
similarly if the method might return null, then mark it as @Nullable